stream/src/main/java/org/apache/kafka/stream/processor/internals/PTopology.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.stream.processor.internals;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.stream.processor.KafkaProcessor;
+import org.apache.kafka.stream.processor.KafkaSource;
+import org.apache.kafka.stream.processor.ProcessorContext;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PTopology {
+
+    private Map<String, KafkaProcessor> processors = new HashMap<>();
+    private Map<String, KafkaSource> topicSources = new HashMap<>();
```

Comment:
Since the key (String) is not the processor name as in processors, but the topic name; i.e. this is really a map from topics to sources.